src/lib.rs

```diff
@@ -15,8 +17,8 @@ pub mod cache;
 mod directory;
 mod error;
 mod header;
-#[cfg(feature = "__async")]
 mod tile;
+mod writer;
```

Comment:
could you clarify why cfg(__async) was removed here?  We really ought to make this PR behind a feature flag

Reply:
I was glad to reduce the `cfg(__)` complexity a bit with 72f57629 - it reminded me of the bad old C/++ `#ifndef` days. IMO, a pmtile library should come with its core functionality (like calculating a tile id) and dependencies (like flate2) by default. I understand the feature flags for the different backend implementations with heavy or specialized dependencies, but not for everything. That said, if the maintainers prefer a feature flag for writing pmtile files, I can give it a try.

Reply:
I agree that we shouldn't go overboard on features.  My only concern is that if we have additional dependencies due to write support, it is better to make write - optional, simply because write is a relatively rare usage compare to read. My understanding is that you need compression, which might be a significant extra compilation burden - and that IMO should not be levied on users unless needed.